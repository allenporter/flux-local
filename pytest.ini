[pytest]
log_level = DEBUG
asyncio_mode = auto
