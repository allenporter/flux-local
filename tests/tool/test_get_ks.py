"""Tests for the flux-local `get ks` command."""

import pytest

from syrupy.assertion import SnapshotAssertion

from . import run_command


@pytest.mark.parametrize(
    ("args"),
    [
        (["--path", "tests/testdata/cluster"]),
        (["--path", "tests/testdata/cluster2"]),
        (
            [
                "-A",
                "--path",
                "tests/testdata/cluster3",
                "--sources",
                "cluster=tests/testdata/cluster3",
            ]
        ),
        (["--path", "tests/testdata/cluster4"]),
        (["--path", "tests/testdata/cluster5"]),
        (["--all", "--path", "tests/testdata/cluster5"]),
        (["--path", "tests/testdata/cluster6"]),
        (["--path", "tests/testdata/cluster7"]),
        (["--path", "./tests/testdata/cluster/clusters/prod"]),
        (["--path", "tests/testdata/cluster", "-o", "wide"]),
        (["--all-namespaces", "--path", "./tests/testdata/cluster/apps/prod"]),
    ],
    ids=[
        "cluster",
        "cluster2",
        "cluster3",
        "cluster4",
        "cluster5",
        "cluster5-all",
        "cluster6",
        "cluster7",
        "cluster_path",
        "wide",
        "ks_path",
    ],
)
async def test_get_ks(args: list[str], snapshot: SnapshotAssertion) -> None:
    """Test test get ks commands."""
    result = await run_command(["get", "ks"] + args)
    assert result == snapshot
