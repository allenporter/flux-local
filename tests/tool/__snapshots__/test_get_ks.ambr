# serializer version: 1
# name: test_get_ks[cluster2]
  '''
  NAME                                       PATH                                                                  
  cluster                                    tests/testdata/cluster2/flux                                          
  cluster-apps                               tests/testdata/cluster2/apps                                          
  cluster-apps-ingress-nginx                 tests/testdata/cluster2/apps/networking/ingress-nginx/app             
  cluster-apps-ingress-nginx-certificates    tests/testdata/cluster2/apps/networking/ingress-nginx/certificates    
  cluster-apps-kubernetes-dashboard          tests/testdata/cluster2/apps/monitoring/kubernetes-dashboard/app      
  
  '''
# ---
# name: test_get_ks[cluster3]
  '''
  NAMESPACE      NAME          PATH                                                    
  flux-system    namespaces    tests/testdata/cluster3/namespaces/overlays/cluster3    
  flux-system    tenants       tests/testdata/cluster3/tenants/overlays/cluster3       
  
  '''
# ---
# name: test_get_ks[cluster4]
  '''
  NAME                                 PATH                                                            
  cluster                              tests/testdata/cluster4/flux                                    
  cluster-apps                         tests/testdata/cluster4/apps                                    
  cluster-apps-kubernetes-dashboard    tests/testdata/cluster4/apps/monitoring/kubernetes-dashboard    
  
  '''
# ---
# name: test_get_ks[cluster5-all]
  '''
  NAMESPACE      NAME                 PATH                                     
  controllers    infra-controllers    tests/testdata/cluster5                  
  flux-system    flux-system          tests/testdata/cluster5/clusters/prod    
  
  '''
# ---
# name: test_get_ks[cluster5]
  '''
  NAME           PATH                                     
  flux-system    tests/testdata/cluster5/clusters/prod    
  
  '''
# ---
# name: test_get_ks[cluster6]
  '''
  NAME           PATH                               
  apps           tests/testdata/cluster6/apps       
  flux-system    tests/testdata/cluster6/cluster    
  
  '''
# ---
# name: test_get_ks[cluster7]
  '''
  NAME           PATH                                     
  apps           tests/testdata/cluster7/flux/apps        
  charts         tests/testdata/cluster7/flux/charts      
  flux-system    tests/testdata/cluster7/clusters/home    
  
  '''
# ---
# name: test_get_ks[cluster9]
  '''
  NAME           PATH                                    
  apps-stack     tests/testdata/cluster9/apps            
  flux-system    tests/testdata/cluster9/clusters/dev    
  
  '''
# ---
# name: test_get_ks[cluster]
  '''
  NAME                 PATH                                                 
  apps                 tests/testdata/cluster/apps/prod                     
  flux-system          tests/testdata/cluster/clusters/prod                 
  infra-configs        tests/testdata/cluster/infrastructure/configs        
  infra-controllers    tests/testdata/cluster/infrastructure/controllers    
  
  '''
# ---
# name: test_get_ks[cluster_path]
  '''
  NAME                 PATH                                                 
  apps                 tests/testdata/cluster/apps/prod                     
  flux-system          tests/testdata/cluster/clusters/prod                 
  infra-configs        tests/testdata/cluster/infrastructure/configs        
  infra-controllers    tests/testdata/cluster/infrastructure/controllers    
  
  '''
# ---
# name: test_get_ks[ks_path]
  '''
  no Kustomization objects found in cluster
  
  '''
# ---
# name: test_get_ks[wide]
  '''
  NAME                 PATH                                                 HELMREPOS    RELEASES    
  apps                 tests/testdata/cluster/apps/prod                     0            1           
  flux-system          tests/testdata/cluster/clusters/prod                 0            0           
  infra-configs        tests/testdata/cluster/infrastructure/configs        3            0           
  infra-controllers    tests/testdata/cluster/infrastructure/controllers    0            3           
  
  '''
# ---
